app/components/global/AppBar/AppBar.test.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import { mount } from 'enzyme'
+import { MockedProvider } from 'react-apollo/test-utils'
+import { ThemeProvider } from 'styled-components'
+import theme from '@elifesciences/elife-theme'
+import { MemoryRouter } from 'react-router-dom'
+// eslint-disable-next-line import/no-named-default
+import { default as AppBarWrapper } from '.'
+import { CURRENT_USER } from '../queries'
+import AppBar from './AppBar'
+
+const defaultProps = {
+  defaultMenuItems: [
+    { label: 'Author guide', link: '/author-guide' },
+    { label: 'Reviewer guide', link: '/reviewer-guide' },
+    { label: 'Contact us', link: '/contact-us' },
+  ],
+  userMenuItems: [{ label: 'Dashboard', link: '/' }],
+}
+
+const mockUserrData = {
+  data: {
+    currentUser: {
+      id: '111',
+      identities: [
+        {
+          name: 'Joe Bloggs',
+          email: '',
+          aff: '',
+          meta: {
+            firstName: 'Joe',
+            lastName: 'Bloggs',
+            __typename: 'ElifeIdentityMeta',
+          },
+          __typename: 'ElifeIdentity',
+        },
+      ],
+      __typename: 'User',
+    },
+  },
+}
+
+const currentUserMock = result => [
+  {
+    request: {
+      query: CURRENT_USER,
+    },
+    result,
+  },
+]
+
+const makeWrapper = (props, mocks = []) =>
+  mount(
+    <ThemeProvider theme={theme}>
+      <MockedProvider addTypename mocks={mocks}>
+        <MemoryRouter>
+          <AppBarWrapper {...props} />
+        </MemoryRouter>
+      </MockedProvider>
+    </ThemeProvider>,
+  )
+
+describe('AppBar', () => {
+  it('Only receives default menu links when no user object passed', async () => {
+    const wrapper = makeWrapper(defaultProps, currentUserMock({}))
+    await new Promise(resolve => {
+      setTimeout(resolve, 0)
+    })
+    wrapper.update()
+    expect(wrapper.find(AppBar).props().menuItems).toHaveLength(3)
```

Comment:
Correct me if I've misunderstood. But this seems like it's testing `index.js` rather than `AppBar.js`?

Specifically this code:
https://github.com/elifesciences/elife-xpub/blob/4e1ca9b68028091362a95d59bb255e30a3aeed7d/app/components/global/AppBar/index.js#L11-L13

because these tests are checking the length of the `menuItems` prop, based on the `user` prop.

If I'm understanding correctly, I imagine that this file should be `index.test.js` rather than `AppBar.test.js`? You can also see this from the fact that [your `makeWrapper` function is building `<AppBarWrapper>`](https://github.com/elifesciences/elife-xpub/pull/927/commits/89cf0b19cf5e35546212306de58a05cde9387e2c#diff-6372abc70d14d6e8c893cb0f7b83fc6fR57)

---

One test that you could write for `AppBar` itself, would be to check whether the appropriate amount of `NavLink`s are rendered, based on props?

But essentially [that mapping](https://github.com/elifesciences/elife-xpub/pull/927/files#diff-db536dc0676835888e436d994badfc32R62) is the only logic built into AppBar itself, so I'm not sure there are actually many tests we could write for this component?